tests/unit/test_netcdf.py

```diff
@@ -547,7 +548,11 @@ def test_nc_with_metadata_fails(scm_run, mdata):
     with tempfile.TemporaryDirectory() as tempdir:
         out_fname = join(tempdir, "out.nc")
 
-        msg = "Invalid value for attr 'test_fails':.*"
+        xarray_version = packaging.version.parse(xr.__version__)
+        if xarray_version >= packaging.version.parse("0.16.2"):
+            msg = "Invalid value for attr 'test_fails':.*"
+        else:
+            msg = "Invalid value for attr: {}".format(str(mdata["test_fails"]))
```

Comment:
Do you need to cast to str? I would have guessed format does that for you

Reply:
Turns out you don't need to..